### ISSUE TITLE
Unable to install v1.0.29 of the extension

### ISSUE THREAD

**Issue by mrchar** [IBM-Blockchain/blockchain-vscode-extension#2349]
The version number of version 1.0.29 is incorrectly marked as 1.0.28，This causes the plugin to update cyclically。

## Description

When I started work this morning, I found that IBP was unavailable, it is cyclically updating. And I found that you updated this plugin 10 hours ago. I guess that the version number marked in the file you released is wrong.

## Possible Fix
fix the released files

## Your Environment
vs-code: 1.45.1 (user setup)
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
IBP: v1.0.28

**Comment by mrchar** [comment]
updating logs blow:
```
[2020-05-22 09:45:39.157] [sharedprocess] [info] Installing extension: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:45:43.211] [sharedprocess] [info] Downloaded extension: ibmblockchain.ibm-blockchain-platform c:\Users\Char\AppData\Local\Temp\42cd3ce6-23ea-44ce-9213-b073debee2a6
[2020-05-22 09:45:43.379] [sharedprocess] [info] Removed the extension from uninstalled list: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:45:45.973] [sharedprocess] [info] Starting to clean up unused language packs.
[2020-05-22 09:45:45.975] [sharedprocess] [info] Skipping directory 2feefd28b9e8042903942873399b35a3.zh-cn. Language pack still in use.
[2020-05-22 09:46:05.254] [sharedprocess] [info] Extracted extension to c:\Users\Char\.vscode\extensions\.ibmblockchain.ibm-blockchain-platform-1.0.28: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:46:05.273] [sharedprocess] [info] Renamed to c:\Users\Char\.vscode\extensions\ibmblockchain.ibm-blockchain-platform-1.0.28
[2020-05-22 09:46:05.281] [sharedprocess] [info] Installation completed. ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:46:05.572] [sharedprocess] [info] Installing extension: oshri6688.javascript-test-runner
[2020-05-22 09:46:05.572] [sharedprocess] [info] Installing extension: ibm.ibmcloud-account
[2020-05-22 09:46:06.059] [sharedprocess] [info] Downloaded extension: ibm.ibmcloud-account c:\Users\Char\AppData\Local\Temp\7ffea6bd-0f1f-4a68-8a9a-3ef1e04bb017
[2020-05-22 09:46:06.060] [sharedprocess] [info] Downloaded extension: oshri6688.javascript-test-runner c:\Users\Char\AppData\Local\Temp\aaeb2f9d-7208-4a2d-8f17-f1c5967ab79a
[2020-05-22 09:46:07.933] [sharedprocess] [info] Extracted extension to c:\Users\Char\.vscode\extensions\.oshri6688.javascript-test-runner-1.2.1: oshri6688.javascript-test-runner
[2020-05-22 09:46:07.936] [sharedprocess] [info] Renamed to c:\Users\Char\.vscode\extensions\oshri6688.javascript-test-runner-1.2.1
[2020-05-22 09:46:07.941] [sharedprocess] [info] Installation completed. oshri6688.javascript-test-runner
[2020-05-22 09:46:07.945] [sharedprocess] [info] Extensions installed successfully: oshri6688.javascript-test-runner
[2020-05-22 09:46:07.985] [sharedprocess] [info] Extracted extension to c:\Users\Char\.vscode\extensions\.ibm.ibmcloud-account-1.0.1: ibm.ibmcloud-account
[2020-05-22 09:46:07.987] [sharedprocess] [info] Renamed to c:\Users\Char\.vscode\extensions\ibm.ibmcloud-account-1.0.1
[2020-05-22 09:46:07.992] [sharedprocess] [info] Installation completed. ibm.ibmcloud-account
[2020-05-22 09:46:07.996] [sharedprocess] [info] Extensions installed successfully: ibm.ibmcloud-account
[2020-05-22 09:46:07.996] [sharedprocess] [info] Extensions installed successfully: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:46:09.534] [sharedprocess] [info] Installing extension: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:46:18.397] [sharedprocess] [info] Downloaded extension: ibmblockchain.ibm-blockchain-platform c:\Users\Char\AppData\Local\Temp\cbbdf862-8d25-4457-acaa-b5ea27cb4d4b
[2020-05-22 09:46:42.168] [sharedprocess] [info] Extracted extension to c:\Users\Char\.vscode\extensions\.ibmblockchain.ibm-blockchain-platform-1.0.28: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:46:42.185] [sharedprocess] [info] Renamed to c:\Users\Char\.vscode\extensions\ibmblockchain.ibm-blockchain-platform-1.0.28
[2020-05-22 09:46:42.192] [sharedprocess] [info] Installation completed. ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:46:42.209] [sharedprocess] [info] Extensions installed successfully: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:46:43.731] [sharedprocess] [info] Installing extension: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:46:47.367] [sharedprocess] [info] Downloaded extension: ibmblockchain.ibm-blockchain-platform c:\Users\Char\AppData\Local\Temp\e56e55dc-e1a2-4111-b626-36dd609480f2
[2020-05-22 09:47:10.657] [sharedprocess] [info] Extracted extension to c:\Users\Char\.vscode\extensions\.ibmblockchain.ibm-blockchain-platform-1.0.28: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:47:10.696] [sharedprocess] [info] Renamed to c:\Users\Char\.vscode\extensions\ibmblockchain.ibm-blockchain-platform-1.0.28
[2020-05-22 09:47:10.704] [sharedprocess] [info] Installation completed. ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:47:10.717] [sharedprocess] [info] Extensions installed successfully: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:47:12.235] [sharedprocess] [info] Installing extension: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:47:15.168] [sharedprocess] [info] Downloaded extension: ibmblockchain.ibm-blockchain-platform c:\Users\Char\AppData\Local\Temp\f4e766b7-a4db-4bb0-bc6b-71bc999f3002
[2020-05-22 09:47:39.190] [sharedprocess] [info] Extracted extension to c:\Users\Char\.vscode\extensions\.ibmblockchain.ibm-blockchain-platform-1.0.28: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:47:39.207] [sharedprocess] [info] Renamed to c:\Users\Char\.vscode\extensions\ibmblockchain.ibm-blockchain-platform-1.0.28
[2020-05-22 09:47:39.215] [sharedprocess] [info] Installation completed. ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:47:39.246] [sharedprocess] [info] Extensions installed successfully: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:47:41.838] [sharedprocess] [info] Installing extension: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:47:44.908] [sharedprocess] [info] Downloaded extension: ibmblockchain.ibm-blockchain-platform c:\Users\Char\AppData\Local\Temp\980a6b87-aca2-40ba-9a33-7142eec33bef
[2020-05-22 09:48:08.556] [sharedprocess] [info] Extracted extension to c:\Users\Char\.vscode\extensions\.ibmblockchain.ibm-blockchain-platform-1.0.28: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:48:08.579] [sharedprocess] [info] Renamed to c:\Users\Char\.vscode\extensions\ibmblockchain.ibm-blockchain-platform-1.0.28
[2020-05-22 09:48:08.588] [sharedprocess] [info] Installation completed. ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:48:08.624] [sharedprocess] [info] Extensions installed successfully: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:48:10.142] [sharedprocess] [info] Installing extension: ibmblockchain.ibm-blockchain-platform
[2020-05-22 09:48:13.146] [sharedprocess] [info] Downloaded extension: ibmblockchain.ibm-blockchain-platform c:\Users\Char\AppData\Local\Temp\30045fa4-0796-44a1-a1e7-f770468c4daf
```

**Comment by sstone1** [comment]
The Visual Studio Code marketplace seems to be in a very confused state; some parts of the marketplace seem to report 1.0.29 is the latest version, and other parts report that 1.0.28 is the latest version - for example: https://vscode-marketplace-api.herokuapp.com/IBMBlockchain/ibm-blockchain-platform

The VSIX we attempted to publish is available here: https://github.com/IBM-Blockchain/blockchain-vscode-extension/releases/download/v1.0.29/ibm-blockchain-platform-1.0.29.vsix

If you install it from that VSIX, then everything is fine, and 1.0.29 works as expected.

I've suggested to the team that we attempt to publish a 1.0.30 this morning and hopefully that will resolve the state of the marketplace.

**Comment by mrchar** [comment]
@sstone1 thank you

**Comment by jnittin** [comment]
Hi There... for me situation looks bit awkward. actually when i open VS code Extension - IBM Blockchain Platform has veriosn 1.0.28 and it was highlighting Alert to install/upgrade to 1.0.29 and while click on 1.0.29 to Install... Its keep on installing mode and not sure how long do i need to wait.
So wondering I'm at right path?

**Comment by Jakeeyturner** [comment]
Hi @jnittin,

We tried publishing a v1.0.30 release earlier however that also failed due to the Azure DevOps service:
```
Publishing IBMBlockchain.ibm-blockchain-platform@1.0.30...
 ERROR  TF10216: Azure DevOps services are currently unavailable. Try again later. Activity Id: 27ab2aeb-cf42-430c-bae4-867cf103ce27
``` 

Other extensions are also seeing the same problem:
- https://github.com/microsoft/vscode/issues/98366
- https://github.com/vscode-contrib/vscode-versionlens/issues/211

Looks like we'll have to wait for Azure to get sorted out first.

Thanks,
Jake

**Comment by jnittin** [comment]
Oh ok.. but you know now i'm having trouble as unable to access 1.0.28 seems 1.0.29 might have been corrupted by 1.0.29 upgrade (just assuming). Now VSCOde extention suggesting me to install IBM BLOCKCHAIN PLATFORM again 1.0.28 and while i clicked on Install .. its more than 30-40 mints still in Installing status? Any suggestion?

**Comment by Jakeeyturner** [comment]
> Oh ok.. but you know now i'm having trouble as unable to access 1.0.28 seems 1.0.28 might have been corrupted by 1.0.29 upgrade (just assuming). Now VSCOde extention suggesting me to install IBM BLOCKCHAIN PLATFORM again 1.0.28 and while i clicked on Install .. its more than 30-40 mints still in Installing status? Any suggestion?

You could try deleting the extension manually - although I expect you won't be able to install it until Azure has fixed the marketplace.
```
rm -rf ~/.vscode/extensions/ibmblockchain.ibm-blockchain-platform-1.0.28/
rm -rf ~/.vscode/extensions/ibmblockchain.ibm-blockchain-platform-1.0.29/
```

Just need to wait for the VS Code marketplace to be fixed, then these problems should go away 🙂

**Comment by emir-garza** [comment]
Thanks, @sstone1 . Installing from the vsix solved it for me.

**Comment by matthewlederman** [comment]
I've posted a possible fix/workaround here.  Hope it helps.

https://github.com/microsoft/vscode/issues/98366

**Comment by mrchar** [comment]
then, I guess this problem is  caused by that vs-code's cdn services are not synchronized. I know nothing about how to maintain a vscode-extension package. but I think there must be some way to control the version, why do I install the v1.0.28 when I install the v1.0.29.

**Comment by Jakeeyturner** [comment]
Just published v1.0.31 - closing this issue as the VS Code marketplace is finally working!
